src/NServiceBus.AcceptanceTests/Routing/When_using_assembly_level_message_mapping_for_pub_sub.cs

```diff
@@ -0,0 +1,115 @@
+﻿namespace NServiceBus.AcceptanceTests.Routing
+{
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using AcceptanceTesting.Customization;
+    using EndpointTemplates;
+    using Features;
+    using NServiceBus.Config;
+    using NUnit.Framework;
+    using ScenarioDescriptors;
+
+    public class When_using_assembly_level_message_mapping_for_pub_sub : NServiceBusAcceptanceTest
+    {
+        static string OtherEndpointName => Conventions.EndpointNamingConvention(typeof(OtherEndpoint));
+
+        [Test]
+        public async Task The_mapping_should_not_cause_publishing_to_non_subscribers()
+        {
+            await Scenario.Define<Context>()
+                .WithEndpoint<OtherEndpoint>()
+                .WithEndpoint<Publisher>(b =>
+                    b.When(c => c.EndpointsStarted, async session =>
+                    {
+                        await session.Publish(new MyEvent());
+                        await session.Send(new DoneCommand());
+                    })
+                )
+                .Done(c => c.CommandReceived || c.EventReceived)
```

Comment:
> Well since the test is about pub sub you would not be sending a command in between but have two endpoints. One subscribed and the other one not.

hm it's a bit a weird scenario since it's not only pubsub since it requires an endpoint mapping for the bug to occur. After the bug is fixed, someone could easily remove the endpoint mapping if we only use subscribers because it would seem like unnecessary. Also, having two subscribers without a command would probably make the test much more vulnerable to flakiness. 

I don't have a strong opinion here. I agree that this test has a weird setup which since it's related to a bug and not a feature but not sure whether the proposed changes improves the situation. Zoom call?